Remove support for `addr`

geoffromer opened the pull request:
Every test that used `addr` before #6283 should be using `ref` after this PR. In most cases that was done in #6283, but this PR transitions a few that I missed in that first pass. In addition, #6283 cloned the old `addr` tests from `foo.carbon` to `foo_addr.carbon` in order to maintain test coverage during the transition; this PR removes those cloned tests.

geoffromer committed:
Remove support for `addr` patterns.

geoffromer committed:
Update examples to use `ref` instead of `addr`

jonmeow reviewed (commented):
This deletes functionality, but should it be making sure the equivalent functionality works with `ref`? What's the intent there?

This may just be a matter of clarifying the PR description, but why is some code migrated to `ref` but then, for example, most of the `addr` class tests are just deleted? Are there equivalent `ref` tests that you can point to, is the coverage no longer relevant for some reason, or is this just disabling support with an intent to re-add it later?

geoffromer commented:
> This deletes functionality, but should it be making sure the equivalent functionality works with `ref`? What's the intent there?
> 
> This may just be a matter of clarifying the PR description, but why is some code migrated to `ref` but then, for example, most of the `addr` class tests are just deleted? Are there equivalent `ref` tests that you can point to, is the coverage no longer relevant for some reason, or is this just disabling support with an intent to re-add it later?

The corresponding `ref` tests were mostly added along with `ref` support in #6283; in general, every deleted test `foo_addr.carbon` (either a physical file or a file split) should have a corresponding `foo.carbon` that's identical except for using `ref` in place of `addr`. While deleting tests in this PR I noticed a few that I missed, so I've added them here to make sure we don't lose any coverage.

jonmeow commented:
> The corresponding `ref` tests were mostly added along with `ref` support in #6283; in general, every deleted test `foo_addr.carbon` (either a physical file or a file split) should have a corresponding `foo.carbon` that's identical except for using `ref` in place of `addr`.

Maybe note this in the PR description? It would've been really helpful for review.

> While deleting tests in this PR I noticed a few that I missed, so I've added them here to make sure we don't lose any coverage.

I don't see any new tests, did you push them?

geoffromer commented:
> > The corresponding `ref` tests were mostly added along with `ref` support in #6283; in general, every deleted test `foo_addr.carbon` (either a physical file or a file split) should have a corresponding `foo.carbon` that's identical except for using `ref` in place of `addr`.
> 
> Maybe note this in the PR description? It would've been really helpful for review.

OK, I've updated the description to try to clarify the mechanics of the transition.

> > While deleting tests in this PR I noticed a few that I missed, so I've added them here to make sure we don't lose any coverage.
> 
> I don't see any new tests, did you push them?

I misspoke; I didn't add any new test files, I just migrated a few files from `addr` to `ref` that I failed to migrate in #6283.

jonmeow commented on toolchain/parse/testdata/generics/generic_params/template_addr.carbon:
```diff
@@ -1,51 +0,0 @@
-// CHECK:STDERR: fn foo(template addr a:! i32);
-// CHECK:STDERR:                 ^~~~
-// CHECK:STDERR:
-fn foo(template addr a:! i32);
```
Approving, though as noted, "template ref" and "ref template" seem not to have tests. That doesn't seem critical to this PR though.

geoffromer replied:
Yeah, good point, I've added those.

jonmeow reviewed (approved):
LG

geoffromer committed:
Respond to reviewer comments

geoffromer committed:
Merge branch 'trunk' into drop-addr
